File: matlab/tagiofuns/get_researcher.m
function		s = get_researcher(initial)

%		s = get_researcher(initial)

s = [] ;
S = read_csv(which('researchers.csv')) ;
if nargin==0,
	for k=1:length(S),
		fprintf(' %s %s\n',S(k).Initial,S(k).Name) ;
	end	
	return
end

% look for S.Initial that matches researcher initial
k = strmatch(initial,{S.Initial},'exact') ;

if isempty(k),
	fprintf(' No entry matching "%s" in researchers file - edit file and retry\n',initial) ;
	return ;
end

if length(k)>1,
	fprintf(' Multiple entries matching "%s" in researchers file:\n',initial) ;
	for kk=1:length(k),
		fprintf(' %d %s\n',kk,S(k(kk)).Name) ;
	end
	n = input(' Enter number for correct researcher... ','s') ;
	n = str2double(n) ;
	if isempty(n) || isnan(n) || n<1 || n>length(k), return ; end
	k = k(n) ;
end

s = S(k) ;

File: Python/tagiofuns/get_researcher.py
def get_researcher(initial=None):
    """
    s = get_researcher(initial)
    
    initial is a string specifying a researcher's initials

    s is a dictionary containing the researcher's details
    
    """
    
    import pandas as pd
    import os
    
    s = {}
    try:
        S = pd.read_csv('./user/researchers.csv')
    except:
        S = pd.read_csv(os.path.join(os.path.dirname(os.path.abspath(__file__)),'researchers.csv'))
    if not initial:
        for k, init in enumerate(S.Initial):
            print(init, S.Name[k])
        return s
    
    # look for S.Initial that matches researcher initial
    k = [i for i,x in enumerate(S.Initial) if x.lower() == initial.lower()]

    if not k:
        print(f" No entry matching {initial} in researchers file - edit file and retry\n")
        return s
    
    if len(k)>1:
        print(f" Multiple entries matching {initial} in researchers file:\n")
        for kk in range(0,len(k)): 
            print(f" {kk} {S.Name[k[kk]]} {S.Email[k[kk]]}")
        n = input(' Enter number for correct researcher... ') 
        n = int(n)
        import math
        if not n or math.isnan(n) or n<0 or n>len(k)-1:
            return s
        k = k[n]
    else:
        k = k[0] 
        
    s = S.iloc[k].to_dict()
    return s